### ISSUE TITLE
Why we need Contributor/Owner role for  SPN in Azure Subscription?

### ISSUE THREAD

**Issue by narendarchandha** [inspec/inspec-azure#418]
General Question: 
I am not sure where to post this question.

We are implementing Chef Inspec for compliance testing for Azure Infra in one of our  Customer.  
They requested a justification of having a service principle with contributor or owner role in the azure subscription as  stated here : https://github.com/inspec/inspec-azure#service-principal 

I have been searching all over the internet, blogs, docs, but nowhere I found justification for this question yet.  
Would you please tell us why we need Contributor/Owner role for my SPN in Azure Subscription?

**Comment by ppradhan9** [comment]
@narendarchandha : We need write access to create azure service principal .The contributor or owner role is appropriate for the same ([supporting doc](https://docs.microsoft.com/en-us/azure/role-based-access-control/rbac-and-directory-admin-roles#azure-roles))

**Comment by sathish-progress** [comment]
Hi @narendarchandha, I did a series of tests with almost all resources(the ones which inspec supports) with the `reader` role at a subscription level and it worked for all resources that inherited from the same subscription and for certain resources I had to add some additional read roles such as `Storage Blob Data Reader` for storage accounts which is expeted. So I could confirm that `reader` role is okay to use this pack. I will update the docs accordingly.
